docs/source/conf.py

```diff
@@ -27,11 +27,12 @@ class DocBuildError(Exception):
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 root_path = os.path.abspath(os.path.join("..", ".."))
 sys.path.insert(0, root_path)
-import commands
-_sphinx_apidoc = commands.getoutput('which sphinx-apidoc').strip()
+from avocado.utils import process
```

Comment:
That's okay, but I'd suggest to import `avocado.utils.process` for the modules in package `avocado_vt` and `virttest` only. The reason to do so is aimed to let these scripts (for example, `scripts/regression.py`, `shared/scripts/cmd_runner.py`) be executed independently with `avocado`. Thus, the best alternative would be `subprocess`.

Reply:
process.run() wrap subprocess and provide more flexible ways to deal with different use case.

Reply:
IMHO, we don't need that "flexible ways" in the user case I described, but we **may** need the flexible porting capability.

Reply:
@luckyh subprocess could not replace the commands totally, only subprocess.check_output could provide the stdout, but the command could not ignore failure.

Reply:
@hsmj1412 try `subprocess.Popen` please.